compiler/passes/convert-uast.cpp

```diff
@@ -3773,8 +3823,28 @@ struct Converter {
     return ret;
   }
 
+  // When converting variables etc. with @assertOnGpu or @blockSize,
+  // we don't just create a DefExpr; we also create an enclosing block which
+  // contains calls to primitives that implement @assertOnGpu and @blockSize.
+  //
+  // This data structure contains pointers to both.
+  struct VariableDefInfo {
+    DefExpr* variableDef;
+    Expr* entireExpr;
+
+    /**
+      Helper for code that calls 'convertVariable' but doesn't expect to handle
+      blocks with additional primitives, which can be introuced by that call
+      for GPU attributes that need to be propagated to init expressions.
+     */
+    Expr* requireDefOnly() const {
+      CHPL_ASSERT(entireExpr == variableDef);
+      return variableDef;
+    }
+  };
```

Comment:
This part might need some buy-in from other dyno folks.

Reply:
Just to close the loop here (and want to make sure that this wasn't hidden by GH), you don't feel the need?

Reply:
Yes, I think this is fine.